dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -502,55 +591,168 @@ private WorkflowVersion setupWorkflowFilesForVersion(String calculatedPath, Trip
+        } catch (YAMLException | ClassCastException | NullPointerException ex) {
+            String msg = "Invalid .dockstore.yml";
+            LOG.info(msg, ex);
+            return null;
+        }
+        for (String filePath: files) {
+            String fileContent = this.readFileFromRepo(filePath, ref.getLeft(), repository);
+            if (fileContent != null) {
+                SourceFile file = new SourceFile();
+                file.setAbsolutePath(filePath);
+                file.setPath(filePath);
+                file.setContent(fileContent);
+                file.setType(DescriptorLanguage.FileType.DOCKSTORE_SERVICE_OTHER);
+                version.getSourceFiles().add(file);
+            } else {
+                // File not found or null
+                LOG.info("Could not find file " + filePath + " in repo " + repository);
+            }
+        }
 
-            version.getSourceFiles().add(dockstoreYml);
+        return version;
+    }
 
-            // Grab all files from files array
+    /**
+     * Pull descriptor files for the given workflow version and add to version
+     * @param ref Triple containing reference name, branch date, and SHA
+     * @param repository GitHub repository object
+     * @param version Version to update
+     * @param workflow Workflow to add version to
+     * @param existingDefaults Existing defaults
+     * @param dockstoreYml Dockstore YML sourcefile
+     * @return Version with updated sourcefiles
+     */
+    private WorkflowVersion setupWorkflowFilesForGitHubVersion(Triple<String, Date, String> ref, GHRepository repository, WorkflowVersion version, Workflow workflow, Map<String, WorkflowVersion> existingDefaults, SourceFile dockstoreYml) {
+        // Determine version information from dockstore.yml
+        Map<String, Object> workflowMap = null;
+        List<String> testParameterPaths = null;
+        try {
             Yaml yaml = new Yaml();
-            List<String> files;
-            try {
-                Map<String, Object> map = yaml.load(dockstoreYmlContent);
-                Map<String, Object> serviceObject = (Map<String, Object>)map.get("service");
-                files = (List<String>)serviceObject.get("files");
-                // null catch due to .dockstore.yml files like https://raw.githubusercontent.com/denis-yuen/test-malformed-app/c43103f4004241cb738280e54047203a7568a337/.dockstore.yml
-            } catch (YAMLException | ClassCastException | NullPointerException ex) {
-                String msg = "Invalid .dockstore.yml";
-                LOG.warn(msg, ex);
-                return version;
+
+            // Find matching workflow entry in .dockstore.yml and grab information
+            Map<String, Object> map = yaml.load(dockstoreYml.getContent());
+            List<Map<String, Object>> workflows = (List<Map<String, Object>>)map.get("workflows");
```

Comment:
Add TODO notes and ticket numbers for #3241
(will skip this going forward, but there are probably many places like this)